crates/db_views/src/post_view.rs

```diff
@@ -368,6 +368,25 @@ impl<'a> PostQueryBuilder<'a> {
         .then_order_by(post_aggregates::stickied.desc());
     }
 
+    // Unless on a community's page or if subscribed to a community, filter out hidden communities from post view
+    if self.community_actor_id.is_none() {
+      if let Some(listing_type) = self.listing_type {
+        query = match listing_type {
+          ListingType::All => query.filter(
+            community::hidden
+              .eq(false)
+              .or(community_follower::person_id.eq(person_id_join)),
+          ),
+          ListingType::Local => query.filter(
+            community::hidden
+              .eq(false)
+              .or(community_follower::person_id.eq(person_id_join)),
+          ),
+          _ => query,
+        };
+      }
+    }
```

Comment:
Instead of this `if self.community_actor_id.is_none()`, do the hiding with the listing_type match on line 357 to be consistent with the same one you had in `comment_view.rs` (hiding this content for all and local). IE

```rust
if let Some(listing_type) = self.listing_type {
      query = match listing_type {
       ...
        ListingType::Local => query.filter(community::local.eq(true)).filter(community::hidden.eq(false)),
       ListingType::All => query.filter.filter(community::hidden.eq(false))
        _ => query,
      };
}
```

Reply:
Done